be2-scala/src/main/scala/ch/epfl/pop/storage/DbActor.scala

```diff
@@ -369,6 +370,21 @@ final case class DbActor(
     (publicKey, privateKey)
   }
 
+  @throws[DbActorNAckException]
+  private def generateHeartbeat(): Option[HashMap[Channel, Set[Hash]]] = {
+    val setOfChannels = getAllChannels
+    if (setOfChannels.isEmpty) return None
+    var heartbeatMap: HashMap[Channel, Set[Hash]] = HashMap()
+    setOfChannels.foreach(channel => {
```

Comment:
I think you can replace this by a setOfChannels.map and avoid the HashMap creation